src/persistence/settings.cpp

```diff
@@ -1379,6 +1409,48 @@ void Settings::setCircleExpanded(int id, bool expanded)
     circleLst[id].expanded = expanded;
 }
 
+void Settings::addFriendRequest(const QString &friendAddress, const QString &message)
+{
+    QMutexLocker locker{&bigLock};
+    QPair<QString, QString> request(friendAddress, message);
+
+    if (friendRequests.indexOf(request) != -1)
+        return;
+
+    friendRequests.push_back(request);
+    ++unreadFriendRequests;
+}
+
+unsigned int Settings::getUnreadFriendRequests() const
+{
+    QMutexLocker locker{&bigLock};
```

Comment:
@TheSpiritXIII Is this actually required on read operations?

Like I wrote in another context, getters/setters should not make use of any mutex in general. This might become a hard to find performance issue. Let me write down two (just logical, not usable) examples, to show what I mean:

```c++
Settings s; // <--- a valied settings object
for (int i = 0; i < 100; ++i) {
     // mutex is activated 100 times, but we won't see it
     // means: stack is copied 100 times 
     s.getUnreadFriendRequests();
     
     // To create more drama, use other getters to read more settings.
     // assumption: Each getter has it's own mutex
     s.getFoo();
     s.getBar();
}
```

Without mutex in getter this would change to this (better) code:

```c++
QMutexLocker(&bigLock);
Settings s; // <--- a valid settings object 
for (int i = 0; i < 100; ++i) {
     // assumption: mutex is NOT set in getter, but above instead
     s.getUnreadFriendRequests();
     s.getFoo();
     s.getBar();
}
```

If I'm not completely wrong, the mutex is accessed just a single time in second example ==> less to worry about ... :smile:

Feel free to prove me wrong (no offence taken). I like to learn for myself.